src/commands/install-packages.yml

```diff
@@ -72,10 +72,11 @@ steps:
                 echo "Found yarn.lock file, assuming lockfile"
                 cp yarn.lock /tmp/node-project-lockfile
               fi
+              cp package.json /tmp/node-project-package.json
         - restore_cache:
             keys:
-              - node-deps-{{ arch }}-<<parameters.cache-version>>-<<#parameters.include-branch-in-cache-key>>{{ .Branch }}-<</parameters.include-branch-in-cache-key>>-{{ checksum "package.json"}}-{{ checksum "/tmp/node-project-lockfile" }}
-              - node-deps-{{ arch }}-<<parameters.cache-version>>-<<#parameters.include-branch-in-cache-key>>{{ .Branch }}-<</parameters.include-branch-in-cache-key>>-{{ checksum "package.json"}}-
+              - node-deps-{{ arch }}-<<parameters.cache-version>>-<<#parameters.include-branch-in-cache-key>>{{ .Branch }}-<</parameters.include-branch-in-cache-key>>-{{ checksum "/tmp/node-project-package.json"}}-{{ checksum "/tmp/node-project-lockfile" }}
+              - node-deps-{{ arch }}-<<parameters.cache-version>>-<<#parameters.include-branch-in-cache-key>>{{ .Branch }}-<</parameters.include-branch-in-cache-key>>-{{ checksum "/tmp/node-project-package.json"}}-
```

Comment:
Partial cache restoration isn't safe on NPM 4- or without a lock file according to https://circleci.com/docs/2.0/caching/#npm-node. I personally think it's reasonable to only support NPM 5+ with a lock file, but if so I would make mention of it in the documentation since this would be a nasty issue to debug.

Reply:
Excellent point! Thank you